packages/medusa/src/services/discount.ts

```diff
@@ -673,14 +675,38 @@ class DiscountService extends TransactionBaseService {
         }, 0)
         const nominator = Math.min(value, subtotal)
         const totalItemPercentage = fullItemPrice / subtotal
-        adjustment = Math.round(nominator * totalItemPercentage)
+
+        // Original adjustment with decimals
+        adjustment = nominator * totalItemPercentage
+
+        const stringifiedAdjustment = adjustment.toString()
+        const pointIndex = stringifiedAdjustment.indexOf(".")
+
+        // Get the big int value of the adjustment
+        adjustment = Number(
+          Math.trunc(adjustment) + stringifiedAdjustment.slice(pointIndex + 1)
+        )
+
+        const zeroCount = stringifiedAdjustment.slice(pointIndex + 1).length
+        multiplierFactor = Number(
+          new Array(zeroCount).fill("0").reduce(function (acc, v) {
+            return acc + v
+          }, "1")
+        )
```

Comment:
Probably it is a good idea to create a helper function for it.

`pointIndex` can be -1 if there is no "." in it.

Reply:
we are doing it only if !== -1. What do you mean?

Reply:
realised that I did not pushed, but we will close the pr in order to think it through. Simple changes does not work as we expected it to arrived and will have other impact that we don't want to rush into too quickly